packages/editor/src/optimizer/worker.ts

```diff
@@ -186,28 +234,26 @@ const startOptimize = (stepSequenceId: StepSequenceID) => {
     return;
   }
 
-  const initVaryingValues = getParentOrCurrentVaryingValues(stepSequenceInfo);
+  const initVaryingValues = varyingValuesSequence.at(-1)!;
   penroseState = {
     ...penroseState!,
     varyingValues: initVaryingValues,
     params: start(initVaryingValues.length),
-    currentStageIndex: stepSequenceInfo.layoutStats.length - 1,
+    currentStageIndex: 0,
   };
 
   penroseState = insertPending(penroseState);
 
   const numStepsPerYield = 1;
-  // on message, we will take a step, but these messages will be queued behind
-  // self.onmessage, so we can interrupt optimization with a message
-  const optStepMsgChannel = new MessageChannel();
 
   // take one optimization step
   const optStep = () => {
     try {
-      if (activeOptimizationId !== stepSequenceId) {
+      if (activeOptimizationId !== startingOptimizationId) {
         // we've been interrupted
```

Comment:
If `startOptimize` is called again, it increments the id and interrupts the old loop. This could happen if we recompile, resample, or interact during optimization.

Reply:
probably good to comment on this